src/domains/pendingProjectAuthorization/repository/pendingProjectAuthorization.repository.impl.ts

```diff
@@ -9,19 +9,35 @@ export class PendingProjectAuthorizationRepositoryImpl implements PendingProject
     const pendingProject: PendingProjectAuthorization | null = await this.db.pendingProjectAuthorization.findFirst({
       where: {
         providerProjectId,
+        deletedAt: null,
       },
     });
 
     return pendingProject === null ? null : new PendingProjectAuthorizationDTO(pendingProject);
   }
 
   async delete(id: string): Promise<PendingProjectAuthorizationDTO> {
-    const pendingProject: PendingProjectAuthorization = await this.db.pendingProjectAuthorization.delete({
+    const deletedPendingProject = this.db.pendingProjectAuthorization.update({
```

Comment:
Please remember to add returned type

Reply:
Didn't add the type because in this case it'll make code hard to read, because it is a Prisma generated type, and if for some reason code changes and type is annotated and not infered, it can cause even more problems.